File: robin/java/src/programmers/level1/study9/GcdLcm.java
package programmers.level1.study9;

import java.util.Arrays;

public class GcdLcm {

    public static int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    public static int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }

    public static int[] solution(int n, int m) {
        int[] answer = new int[2];

        answer[0] = gcd(n, m);
        answer[1] = lcm(n, m);

        return answer;
    }

    public static void main(String[] args) {
        int n = 2;
        int m = 5;
        System.out.println(Arrays.toString(solution(n, m)));
    }
}

File: robin/python/programmers/level1/study9/gcd_lcm.py
def solution(n, m):
    n_copy = n
    m_copy = m
    answer = []
    # gcd
    while True:
        temp = m % n

        if temp == 0:
            answer.append(n)
            break
        m = n
        n = temp

    # lcm
    answer.append(int((n_copy * m_copy) / answer[0]))

    return answer


def gcd(a, b):
    return b if a % b == 0 else gcd(b, a % b)


def lcm(a, b):
    return int(a * b / gcd(a, b))


def gcdlcm(a, b):
    answer = [gcd(a, b), lcm(a, b)]

    return answer


if __name__ == '__main__':
    n = 2
    m = 5
    print(solution(n, m))
    print(gcdlcm(n, m))
